### PR TITLE
#321 Validate MIDI note range

### PR DESCRIPTION
**Pull request by KartikYZ**
Added validation for midi note range in note.go. We raise a UserFacingError if midi note is > 0 or < 127 and log the event.

### CODE REVIEW

File: client/model/note.go [hunk 1]
```diff
@@ -125,6 +126,14 @@ func addNoteOrRest(score *Score, noteOrRest ScoreUpdate) error {
 					part.Octave, part.KeySignature, part.Transposition,
 				)
 
+				if midiNote < 0 || midiNote > 127 {
+					log.Debug().
+						Int("midiNote", int(midiNote)).
+						Msg("Midi note out of range.")
```
Comment (daveyarwood) [inline client/model/note.go:132]: This debug log has the same information as the user-facing error message, so I think we can remove it to reduce noise.

**Review by daveyarwood — COMMENTED** [review]
This looks good, thanks for the PR! My two suggestions would be to remove the debug log message, and to add tests. I saw you were asking about how to approach writing tests on Slack, so I'll answer there.

**Comment by daveyarwood** [comment]
Fixes #321.

File: client/model/note.go (hunk 1 above)
Comment (KartikYZ) [reply client/model/note.go:132]: Thanks for the feedback, I've checked the Slack. Working on it!

**Comment by daveyarwood** [comment]
Superceded by #461

**Pull request closed without merging**